Source/AliveLib/MainMenu.cpp

```diff
@@ -1672,21 +1672,271 @@ void MainMenuController::tGame_BackStory_Or_NewGame_Unload_4D1BE0()
     pResourceManager_5C1BB0->LoadingLoop_465590(FALSE);
 }
 
+char MainMenuController::sub_convert_iso_FileName_4D1660(char *a1, char *a2)
+{
+    char result = *a2;
+    if (*a2)
+    {
+        do
+        {
+            if (result == ';')
+            {
+                break;
+            }
+            ++a2;
+            *a1 = (char) tolower(result);
+            result = *a2;
+            ++a1;
+        } while (*a2);
+        *a1 = 0;
+    }
+    else
+    {
+        result = *a1;
+        *a1 = 0;
+    }
+    return result;
+}
 
-signed int MainMenuController::tsub_LoadSave_Input_4D1040(DWORD /*input*/)
+char MainMenuController::sub_FileName_4D1430(char* input)
 {
-    NOT_IMPLEMENTED();
+    if (input[0] == '\\')
+    {
+        input++;
+    }
+    char buffer[256];
+    MainMenuController::sub_convert_iso_FileName_4D1660(buffer, input);
 
-    return 0;
+    char fileName[256];
+    strcpy(fileName, sCdEmu_Path1_C14620);
+    strcat(fileName, buffer);
+    if (!access_impl(fileName, 0))
+    {
+        return 1;
+    }
+    strcpy(fileName, sCdEmu_Path2_C144C0);
+    strcat(fileName, buffer);
+    if (!access_impl(fileName, 0))
+    {
+        return 1;
+    }
+
+    if (!sCdRomDrives_5CA488[0])
+    {
+        return 0;
+    }
+    strcpy(fileName, sCdEmu_Path3_C145A0);
+    strcat(fileName, buffer);
+    fileName[0] = sCdRomDrives_5CA488[0];
+
+    while (access_impl(fileName, 0))
+    {
+        fileName[0] = (sCdRomDrives_5CA488++)[1];
+        if (!fileName[0])
+        {
+            return 0;
+        }
+    }
+    if (fileName[0])
+    {
+        return 1;
+    }
+    else
+    {
+        return 0;
+    }
 }
 
-signed int MainMenuController::tsub_4D0E10(DWORD /*input*/)
+const __int16 word_5617F4[4] = {
+    1, 8, 5, 0
+};
+
+ALIVE_VAR(1, 0x5C1B9C, short, word_5C1B9C, 0);
+ALIVE_VAR(1, 0x5C1BA2, BYTE, byte_5C1BA2, 0);
+ALIVE_VAR(1, 0x55C128, int, dword_55C128, 0);
+
+signed int MainMenuController::tLoadDemo_4D1040(DWORD)
 {
-    NOT_IMPLEMENTED();
+    const int maxDemoId = 23;
+
+    if (field_23C_T80.Get(Flags::eBit18_Loading))
+    {
+        __int16 demoId = word_5C1B9E;
+        if (word_5C1B9C == 0)
+        {
+            demoId = byte_5C1BA2;
+        }
+        if (demoId >= maxDemoId)
+        {
+            demoId = 0;
+        }
+        int levelId = static_cast<int>(sDemos_5617F0[demoId].field_4_level);
+        char lvFilename[256];
+        strcpy(lvFilename, "ATTRACT");
```

Comment:
this buffer looks like 2 local arrays? one with ATTRACT in and one with the lvl name ??

Reply:
One is a part of the other, resolved by adding lvFilenameNoPrefix